cli/pawls/commands/dataset.py

```diff
@@ -0,0 +1,51 @@
+import os
+import glob
+import click
+import shutil
+import hashlib
+
+from tqdm import tqdm
+from pathlib import Path
+from typing import Union
+
+
+def hash_pdf(file: Union[str, Path]) -> str:
+    block_size = 65536
+
+    file_hash = hashlib.sha256()
+    with open(str(file), 'rb') as fp:
+        fb = fp.read(block_size)
+        while len(fb) > 0:
+            file_hash.update(fb)
+            fb = fp.read(block_size)
+
+    return str(file_hash.hexdigest())
+
+
+def copy(source: Union[str, Path], destination: Union[str, Path]) -> None:
+    shutil.copy(str(source), str(destination))
+
+
+@click.command(context_settings={"help_option_names": ["--help", "-h"]})
+@click.argument("directory", type=click.Path(exists=True, file_okay=True, dir_okay=True))
+def add(directory: click.Path) -> None:
+    """
+    Add a PDF or directory of PDFs to the pawls dataset (skiff_files/).
+    """
+    base_dir = Path('skiff_files/apps/pawls/papers')
+    base_dir.mkdir(exist_ok=True, parents=True)
+
+    if os.path.isdir(str(directory)):
+        pdfs = glob.glob(os.path.join(str(directory), '*.pdf'))
+    else:
+        pdfs = [str(directory)]
+
+    print(f'Found {len(pdfs)} total PDFs to add.')
+
+    for pdf in tqdm(pdfs):
+        sha = hash_pdf(pdf)
```

Comment:
We may want to let the client choose to preserve the filename, rather than always using a hash.

For instance, in our case, the PDFs are usually already given a name that's a hash of the content. Though in our case that hash is produced by a separate system that's responsible for clustering PDFs.

@lolipopshock WDTY?

Reply:
I was thinking that the `add` command can be run numerous times, so you can't guarantee names (for different documents) won't collide between runs.

Maybe a `--preserve-filename` arg and a `logging.warning(f'Skipping {pdf}')` if there's a name collision? On that note, I should probably add a warning if there's an existing hash for a PDF you're trying to add.